src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -100,7 +100,11 @@
                         ;; Try to get a connection to make sure the db is ready
                         (.close (.getConnection datasource))
                         (when-let [{:keys [migration-db migration-dir replication-mode]} migration-opts]
-                          (when-not (= replication-mode :replica)
+                          (if (= replication-mode :replica)
+                            (loop []
+                              (when (not-empty (migration/uncompleted-migrations migration-db migration-dir))
+                                (Thread/sleep 1500)
```

Comment:
Can we declare this magic number somewhere in this file (with a name that documents the units) and use it here?  How confident are we that this is the right number?  Is this something that we should be able to override via the migration opts?